src/grisbi_osx.m

```diff
@@ -58,19 +58,28 @@
     return program_name;
 }
 
-static char const *get_program_dir()
+static gchar *get_program_dir()
 {
-    return g_path_get_dirname(get_program_name());
+    char *program_name = get_program_name();
+    gchar *program_dir = g_path_get_dirname(program_name);
+    free(program_name);
+    return program_dir;
 }
 
 static gchar *get_bundle_prefix(void) {
-    char const *program_dir = get_program_dir();
-
+    gchar *get_bundle_prefix;
+    gchar *program_dir = get_program_dir();
     gchar *prefix = g_path_get_dirname(program_dir);
+
     if (g_str_has_suffix(program_dir, "Contents/MacOS")) {
-        prefix = g_build_filename(prefix, "Resources", NULL);
+        get_bundle_prefix = g_build_filename(prefix, "Resources", NULL);
+    } else {
+        get_bundle_prefix = g_strdup(prefix);
     }
 
+    g_free(program_dir);
+    g_free(prefix);
+
```

Comment:
Do not `g_free(prefix)` just before you return it.

Reply:
I means return bundle_prefix.... FIXED